tests/test_handlers.py

```diff
@@ -0,0 +1,172 @@
+                                 '{bold_yellow}\d{{4}}-\d{{2}}-\d{{2}} \d{{2}}:\d{{2}}:\d{{2}},\d+ '
+                                 'FRONTIER_START   info message{reset}\n'.format(
+                                     bold_yellow=re.escape(colors['bold_yellow']),
+                                     reset=re.escape(colors['reset'])))
+
+    def test_handler_events_level(self):
+        self.logger.debug('debug message', extra={'event': 'FRONTIER_START'})
+        self.logger.info('info message', extra={'event': 'FRONTIER_STOP'})
+        self.assertRegexpMatches(self.logger_output.getvalue(),
+                                 '{bold_yellow}\d{{4}}-\d{{2}}-\d{{2}} \d{{2}}:\d{{2}}:\d{{2}},\d+ '
+                                 'FRONTIER_STOP    info message{reset}\n'.format(
+                                     bold_yellow=re.escape(colors['bold_yellow']),
+                                     reset=re.escape(colors['reset'])))
+
+    def test_handler_events_filter(self):
+        self.logger.info({'message1': 'info', 'message2': 'message', 'event': 'FRONTIER_START'})
+        log = self.logger_output.getvalue()
+        pattern1 = '{bold_yellow}\d{{4}}-\d{{2}}-\d{{2}} \d{{2}}:\d{{2}}:\d{{2}},\d+ ' \
+                   'FRONTIER_START   {logged_message}{reset}\n'.format(bold_yellow=re.escape(colors['bold_yellow']),
+                                                                       logged_message='message info',
+                                                                       reset=re.escape(colors['reset']))
+        pattern2 = '{bold_yellow}\d{{4}}-\d{{2}}-\d{{2}} \d{{2}}:\d{{2}}:\d{{2}},\d+ ' \
+                   'FRONTIER_START   {logged_message}{reset}\n'.format(bold_yellow=re.escape(colors['bold_yellow']),
+                                                                       logged_message='info message',
+                                                                       reset=re.escape(colors['reset']))
+        assert re.match(pattern1, log) is not None or re.match(pattern2, log) is not None
+
+
+class TestHandlerConsole(BaseTestHandlers):
+    handler = CONSOLE
+
+    def test_handler_color_based_on_level(self):
+        self.logger.debug('debug message')
+        self.logger.info('info message')
+        self.logger.error('error message')
+        self.assertEqual(self.logger_output.getvalue(),
+                         '{white}[frontera] debug message{reset}\n'
+                         '{green}[frontera] info message{reset}\n'
+                         '{red}[frontera] error message{reset}\n'.format(white=colors['white'],
+                                                                         green=colors['green'],
+                                                                         red=colors['red'],
+                                                                         reset=colors['reset']))
+
+
+class TestHandlerConsoleManager(TestHandlerConsole):
```

Comment:
Please check that _MANAGER, _BACKEND, _DEBUGGING  and _EVENTS is used somewhere. As I remember they don't. May be it's easier to remove this handlers than writing a tests?

Last time when I was refactoring the logging system, I switched to per-class loggers and their configuration is meant to be done by means of ``logging.config`` package (e.g. adding handlers). The handlers above and other components I left because they appeared useful to me, but **I never used them**, and not aware of somebody else using them. Now, the biggest question for me as a reviewer: does it make sense to test them, or better to remove them?

If we leave them, then yes we need tests and much much more we need a document explaining they exists and how to use them.

Color logging is nice feature, but useful only when running in CMD line. JSON formatter is useful for sure. Filters? Probably yes, mainly to filter events by type, or URLs. Redis? Do you think it's a good idea to store log in redis? Why not setup a local logging to file with rollback? Overall, it needs thought.

I don't want to demotivate you, but this is how it is.